### DISCUSSION TITLE
Be able to import .dsl files?

### DISCUSSION THREAD

**Discussion by psacchitella** [likec4/likec4#308] [Ideas]
Hello,
Is there any plan to be able to import existing .dsl files? We have been using the Structurizr Dashboard to host our diagrams. I would like to switch to something lighter and faster but we have over 100 .dsl files in our workspace. 

Is there any plan to create a feature that allows .dsl file imports or conversions?

Cheers,
Phil

**Comment by davydkov** [comment]
Hi @psacchitella 

Wow, over 100 files!! 😮 

The initial idea is to make the syntax close to Structurizr DSL syntax, exactly for the purposes of more-or-less smooth migration (as I had Structurizr project myself). Not everything is supported. Especially the views require modifications.

I want to say, at the moment, there is no automatic conversion. I did it manually.
Sure, for the 100 files, this does not sound appealing and I want to help you.

I have not thought about automatic conversion, maybe something  
Structurizr DSL => Structurizr JSON => LikeC4 JSON => LikeC4 DSL

What do you think?

**Reply by JasonBSteele** [reply]
I would love to see this too! I don't have 100 files, but I do have about 25 - some of them quite big.

**Reply by psacchitella** [reply]
I am sorry, for some reason I didn't see your reply.
That would be great. Although right now we have switched to IcePanel because the on-prem structurizr instance was so twitchy. I also couldn't get many of the teams to create the DSL files.
Keep up the good work though.
Cheers,
Phil

**Reply by davydkov** [reply]
Hey @psacchitella 

Sorry for the delay in getting back to you!

My focus has shifted a bit towards working on the new renderer and extending DSL, so I haven't had a chance to think much about the import feature.

IcePanel is indeed a great product! I remember using it when they were just starting out, and even back then, it was already quite mature.
However, I still prefer coding over visual tools.

**Comment by Kiiv** [comment]
Hi!

I've started with Structurizr since a few month before discovering this tool. The UI seems really appealing and some features like includes options seems much better. I think to migrate to LikeC4 but I also have a looot of files and elements making me think twice before I go :p
This kind of feature could really help to convince people in this situation.

**Comment by escar** [comment]
This is somewhat old, but we have recently done this using AI. It worked out pretty well.

While Structurizr can read .dsl files in runtime, LikeC4 bundles all files in javascript. Are there any performance concerns in bundling too many files? I'm wondering if you tried this with your 100+ files @psacchitella.
